src/blog/public_vs_private_state.md

```diff
@@ -0,0 +1,77 @@
+
+Zcash was one of the earliest chains to utilize the power of zero knowledge proofs to provide privacy to an account's balance. Unlike traditional cryptocurrencies like Bitcoin, where transaction details are publicly visible on the blockchain, Zcash enables users to shield their transactions, rendering them completely opaque to outside observers. It utilizes zk-SNARKs (Zero-Knowledge Succinct Non-Interactive Argument of Knowledge), a type of zero-knowledge proof system that allows parties to verify the validity of a computation without revealing the underlying data. 
+
+In even more recent times, we have seen several Ethereum virtual machine-compatible Layer-2 chains that have employed zero-knowledge cryptography techniques to solve Ethereum's scalability issues. Polygon zkEVM, zkSync and Starknet, for example, are ZK rollups which have successfully reduced transaction fees significantly. All of them utilize zero-knowledge technology for scalability reasons.
+
+Aleo is a new layer-1 blockchain that combines general-purpose programmability with privacy by default. Unlike the other chains, Aleo is one of the first blockchains to utilize zero-knowledge for both privacy and scalability. 
+
+
+## Privacy
+There are generally four different types of privacy that relate to blockchains, of which Aleo fulfils the 3 of them:
+- [x] Private inputs (messages)
+- [x] Private outputs (state changes)
+- [x] Private user
+- [ ] Private function 
+
+The core belief at Aleo is that privacy is necessary in order for blockchains to have mainstream adoption. For instance, how many of us would be comfortable revealing our bank account numbers if anyone in public were able to trace all the transactions we have ever made? 
+
+## Aleo State Storage 
+In order to have privacy native to a chain, Aleo uses a record model for application state storage, which is similar to the UTXO model in Bitcoin. However, Aleo provides developers the option to make application states public should they choose to do so. Public states are stored using the account model as done in Ethereum. 
+
+### Storing Private States via Record
+Records are a fundamental data structure that can contain any arbitrary payload and are used for encoding user assets or application states. A record represents a certain state of a program in the global state, for example, the balance of an account or your identity document. 
+
+
+An Aleo record is serialized in the following format:
+
+| Parameter  |             Type             | Description                                                                                         |
```

Comment:
The record structure has been updated and now it is slightly different from the current documentation. I think using the [updated version](https://github.com/Aleo-DevRel-Ambas/zero-to-zk/blob/blog/public_vs_private_state/src/learn/concepts/storage.md#components-of-a-record) is better for the unity of the content.

Reply:
I used 'arbitrary' for the data type because it was suggested by Ramiro in another article that 'modifiable' implies that the record can be modified, whereas it can only be created and destroyed.

Reply:
I think I missed that. If the majority accepts that opinion, I should revise mine. I'll ask about this on Telegram. Thanks for pointing it out!

Reply:
@ramidecodes Hey Ramiro! I contacted you on Telegram but forgot to tag you here. This is the part that I mentioned in my message

Reply:
Well, currently Record `data` property is defined as a `map` type in the official docs. So we can describe it with the same `type` here also. So no `arbitrary` or `mutable`, but `Map`
And regarding the description, I still believe that "arbitrary" is a better word to describe the intention of the `data` property, since it's up to the user to decide what to store here. If we describe it as mutable it implies that after defining an initial value for it user could update that value after the record has been created. In reality you can only consume or destroy a record.

https://developer.aleo.org/concepts/records/

Reply:
It is stated in the rest of the docs that a record is either created or consumed, but you are right that it may cause confusion at first glance. Although I think the map is not perfectly clear, developers will probably check out the rest to understand the concept. My personal opinion is, changing both of them to `map` will be more helpful. I'll write it on telegram group for further clarity. Thanks for the help!

Reply:
Yes I agree both should be changed to `Map`.

Reply:
I updated previously merged PR in [here](https://github.com/Aleo-DevRel-Ambas/zero-to-zk/pull/27)

Reply:
Ok, I updated to map as well.